users/views.py

```diff
@@ -1,3 +1,143 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect, get_object_or_404
+from django.contrib.auth import authenticate
+from django.contrib.auth import login as auth_login
+from django.contrib.auth import logout as auth_logout
+from django.urls import reverse
+from .models import CustomUser, Skill
 
-# Create your views here.
+
+def login(request):
+    if request.user.is_authenticated:
+            return render(request, 'users/test.html', {})
+    else:
+        if request.method == 'POST':
+            username = request.POST.get('username', '')
+            password = request.POST.get('password', '')
+            user = authenticate(username=username, password=password)
+            if user:
+                if user.is_active:
+                    auth_login(request, user)
+                    return render(request, 'users/test.html', {})
+                else:
+                    error = 'The account has been disabled.'
+                    return render(request, 'users/login.html',
+                                  {'error': error})
+            else:
+                error = 'Invalid Username/Password'
+                return render(request, 'users/login.html', {'error': error})
+        else:
+            return render(request, 'users/login.html', {})
+
+
+def logout(request):
+    auth_logout(request)
+    return redirect(reverse('users:login'))
+
+
+def register(request):
+    if request.user.is_authenticated:
+        return render(request, 'users/test.html', {})
+    else:
+        if request.method == 'POST':
+            # Username and email fields will have the same data
+            username = request.POST.get('email', '')
+            password = request.POST.get('password', '')
+            sap_id = request.POST.get('sap_id', '')
+            mobile = request.POST.get('mobile')
+            # Check if no other user has the same email id
+            if CustomUser.objects.filter(username=username).exists():
+                error = 'The email is already in use by another account.'
+                return render(request, 'users/register.html', {'error': error})
+            # Check for uniqueness of SAP ID
+            elif CustomUser.objects.filter(sap_id=sap_id).exists():
+                error = 'The SAP ID is already in use by another account.'
+                return render(request, 'users/register.html', {'error': error})
+            # Check for uniqueness of Mobile No.
+            elif CustomUser.objects.filter(mobile=mobile).exists():
```

Comment:
Instead of naming the error as error always, name it error1, error2, error3 such that you'll be able to distinguish them on the front end.